File: Java/DecoratorPattern/Coffee-Hut/src/com/laxmena/beverage/Beverage.java
package com.laxmena.beverage;

/**
 * Beverage Abstract Class.
 *
 * Other Concrete Beverage classes like Dark Roast, Latte, and Expresso
 * extend this base class.
 */
public abstract class Beverage {
    String description = "Some Beverage!";

    public String getDescription() {
        return this.description;
    }

    public abstract double cost();
}

File: Decorator/Python/Coffee-Hut/beverage/beverage.py
from abc import ABC, abstractmethod


class Beverage(ABC):
    """Abstract Class that all Concrete Beverage class should implement.

    Beverage class contains two methods and an attribute defined.
    cost is an abstract method that the child classes should implement.
    """
    description = "Unnamed Beverage"

    def get_description(self) -> str:
        return self.description

    @abstractmethod
    def cost(self) -> float:
        pass
